src/dataplane-protocol/src/batch.rs

```diff
@@ -27,7 +24,17 @@ pub trait BatchRecords: Default + Debug + Encoder + Decoder {
     }
 }
 
-impl BatchRecords for DefaultBatchRecords {}
+#[allow(deprecated)]
+#[deprecated(since = "0.6.0", note = "Use VecBatch instead")]
+pub type DefaultBatchRecords = Vec<crate::record::DefaultRecord>;
+
+#[allow(deprecated)]
+#[deprecated(since = "0.6.0", note = "Use plain Batch instead")]
+pub type DefaultBatch = Batch<DefaultBatchRecords>;
+
+pub type VecBatch = Vec<Record>;
```

Comment:
I would prefer name as something like`MemoryBatch`,'BytesBatch` or something like that to connote that this Records is represented in Memory not in file slice or some other format.   `Vec` is just current implementation but should not assume that.  We should discourage to rely on Vec

Reply:
I like `MemoryBatch` 👍